packages/dev/core/src/Gizmos/scaleGizmo.ts

```diff
@@ -339,26 +372,29 @@ export class ScaleGizmo extends Gizmo implements IScaleGizmo {
      * Disposes of the gizmo
      */
     public dispose() {
-        [this.xGizmo, this.yGizmo, this.zGizmo, this.uniformScaleGizmo].forEach((gizmo) => {
+        const allGizmos = this.allGizmos;
+        const gizmoCount = allGizmos.length;
+        for (let i = 0; i < gizmoCount; i++) {
+            const gizmo = allGizmos[i];
             if (gizmo) {
                 gizmo.dispose();
             }
-        });
-        this._observables.forEach((obs) => {
-            this.gizmoLayer.utilityLayerScene.onPointerObservable.remove(obs);
-        });
+        }
+        const { _observables, _uniformScalingMesh, _octahedron } = this;
+        const _obserLength = _observables.length;
+        const onPointerObservable = this.gizmoLayer.utilityLayerScene.onPointerObservable;

```

Comment:
Please remove all this and simply prefix with `this.` or `this.gizmoLayer.utilityLayerScene.onPointerObservable.` where appropriate.

The new code is unquestionably less readable, and you won't get any performance gains by doing so, in the context of typical uses of this class.